src/components/templates/DataTable/TestingTable.tsx

```diff
@@ -0,0 +1,356 @@
+import { FC, useState, HTMLProps, useEffect, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import DataTable, {
+  TableSizeTypes,
+} from 'components/templates/DataTable/DataTable'
+
+import { createColumnHelper, PaginationState } from '@tanstack/react-table'
+import TableFilter from 'components/organisms/TableFilter/TableFilter'
+import Button, {
+  AppearanceTypes,
+  SizeTypes,
+  IconPositioningTypes,
+} from 'components/molecules/Button/Button'
+import { ReactComponent as ExpandIcon } from 'assets/icons/expand.svg'
+import { ReactComponent as ArrowRight } from 'assets/icons/arrow_right.svg'
+import { Link } from 'react-router-dom'
+
+type Person = {
+  id: string
+  firstName: string
+  lastName: string
+  age: number
+  visits: number
+  status: string
+  progress: number
+  subRows?: Person[]
+}
+
+function IndeterminateCheckbox({
+  indeterminate,
+  className = '',
+  ...rest
+}: { indeterminate?: boolean } & HTMLProps<HTMLInputElement>) {
+  const ref = useRef<HTMLInputElement>(null!)
```

Comment:
**[blocker]**

typescript error here, you can fix this, by removing the "null" from useRef constructor and using useImperativeHandle for populating the values of ref.current with some custom values. (example usage inside organisms/modals/index)